test/e2e/cluster_extension_install_test.go

```diff
@@ -82,7 +82,7 @@ func TestClusterExtensionInstallRegistry(t *testing.T) {
 	t.Log("By eventually reporting a successful resolution and bundle path")
 	require.EventuallyWithT(t, func(ct *assert.CollectT) {
 		assert.NoError(ct, c.Get(context.Background(), types.NamespacedName{Name: clusterExtension.Name}, clusterExtension))
-		assert.Len(ct, clusterExtension.Status.Conditions, 8)
+		assert.Len(ct, clusterExtension.Status.Conditions, 7)
```

Comment:
Not a comment on your need to change this, you need to, but this is bad for a couple reasons. 1) brittle test, 2) magic number in the code. 
Would we want to do something better at this point, maybe just checking for some of the Conditions having the statuses we care about rather than their count?

Reply:
> Not a comment on your need to change this, you need to, but this is bad for a couple reasons. 1) brittle test, 2) magic number in the code.

Yes, I agree with this. But on the other hand - this line helped me spot the issue.

I think I can probably replace with something like 

```go
assert.Len(ct, clusterExtension.Status.Conditions, len(conditionsets.ConditionTypes))
```

Reply:
seems decent.